tests/testthat/test-independent-ahr_blinded.R

```diff
@@ -0,0 +1,161 @@
+test_that("ahr_blinded throws an error for non-numeric or negative hr", {
+  expect_error(
+    ahr_blinded(hr = c(1, -2, 3)),
+    "'hr' must be a vector of positive numbers."
+  )
+  expect_error(
+    ahr_blinded(hr = "abc"),
+    "'hr' must be a vector of positive numbers."
+  )
+})
+
+test_that("ahr_blinded throws an error when intervals and hr are not aligned", {
+  expect_error(
+    ahr_blinded(intervals = c(3, 6), hr = c(1)),
+    "The piecewise model specified 'hr' and 'intervals' differ in length."
+  )
+})
+
+test_that("ahr_blinded handles piecewise exponential model fitting and calculations correctly", {
+  surv <- survival::Surv(
+    time = simtrial::ex1_delayed_effect$month,
+    event = simtrial::ex1_delayed_effect$evntd
+  )
+  intervals <- c(3, 6, Inf)
+  hr <- c(1, 0.7, 0.5)
+  ratio <- 2
+
+  # Run the function
+  result <- ahr_blinded(surv = surv, intervals = intervals, hr = hr, ratio = ratio)
+
+  # Test 1: Correct fitting of survival data into piecewise exponential model
```

Comment:
I am wondering if "Test 1" is informative or not... it looks like it does not compare the output of `ahr_blinded` with some double-programming/independent programming results...